kmk/extensions/ble_split.py

```diff
@@ -0,0 +1,195 @@
+'''Enables splitting keyboards wirelessly'''
+from adafruit_ble import BLERadio
+from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
+from adafruit_ble.services.nordic import UARTService
+from kmk.extensions import Extension
+from kmk.hid import HIDModes
+from kmk.kmktime import ticks_diff, ticks_ms
+from kmk.matrix import intify_coordinate
+from storage import getmount
+
+
+class BLE_Split(Extension):
+    '''Enables splitting keyboards wirelessly'''
+
+    def __init__(
+        self, split_flip=True, split_side=None, uart_interval=30, hid_type=HIDModes.BLE
+    ):
+        self._is_target = True
+        self._uart_buffer = []
+        self.hid_type = hid_type
+        self.split_flip = split_flip
+        self.split_side = split_side
+        self.split_offset = None
+        self._ble = BLERadio()
+        self._ble_last_scan = ticks_ms() - 5000
+        self._is_target = True
+        self._connection_count = 0
+        self._uart = None
+        self._uart_connection = None
+        self._advertisment = None
+        self._advertising = False
+        self._uart_interval = uart_interval
+        self._psave_enable = False
+
+    def __repr__(self):
+        return f'BLE_SPLIT({self._to_dict()})'
+
+    def _to_dict(self):
+        return f'''
+        BLE_Split( _ble={self._ble}
+        _ble_last_scan={self._ble_last_scan}
+        _is_target={self._is_target}
+        _uart_buffer={self._uart_buffer}
+        _split_flip={self.split_flip}
+        _split_side={self.split_side} )
+        '''
+
+    def on_runtime_enable(self, keyboard):
+        return
+
+    def on_runtime_disable(self, keyboard):
+        return
+
+    def during_bootup(self, keyboard):
+        self._ble.name = str(getmount('/').label)
+        if self._ble.name.endswith('L'):
```

Comment:
While I would agree that it's rare, it's also about impossible to tell "which is which" in the case of split keyboards. That is what left me to rename them, document it in the guide, and offer it as an option. I didn't want to FORCE the opinion, so if you manually use split_side it will override the KLAR name. If not, well, it wasn't working anyways as you didn't give a split_side.

Reply:
I was incorrect. Patch incoming to make that statement correct. split_side priority, fallback to last letter.